packages/parser/src/language/sankey/sankeyMatcher.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Matches sankey link source node
+ */
+export const sankeyLinkSourceRegex = /(?:"((?:""|[^"])+)")|([^\n\r,]+)/;
+
+/**
+ * Matches sankey link target node
+ */
+export const sankeyLinkTargetRegex = /,(?:(?:"((?:""|[^"])+)")|([^\n\r,]+))/;
+
+/**
+ * Matches sankey link value
+ */
+export const sankeyLinkValueRegex = /,("(?:0|[1-9]\d*)(?:\.\d+)?"|[\t ]*(?:0|[1-9]\d*)(?:\.\d+)?)/;
```

Comment:
It is not clear why this complex regex is necessary. It feels like there should be an easier way.
When comparing with the deleted `.jison` file, the langium one is a lot more complex. Which makes it harder for people to create new diagrams. 

Something similar is duplicated inside the `.langium` file too. 
```
terminal SANKEY_LINK_VALUE returns number: /,("(0|[1-9][0-9]*)(\.[0-9]+)?"|[\t ]*(0|[1-9][0-9]*)(\.[0-9]+)?)/;
terminal SANKEY_LINK_TARGET: /,("((?:""|[^"])+)"|[^\n\r,]+)/;
```

Reply:
Matchers (most if the time) are just regexes to extract wanted data from tokens.

So here in these matcher regexes, it should match what is between doable quotes without including the wrapper doable quotes or everything until `,`.
It could be simplified and replace the first and last char, but we'll need to make sure which pattern has been matched.

---

As for `.langium`, I'm not sure if we could simplify it without allowing weird pattern.
So in the value regex, disallowed some patterns:

```
, "0" %% no space after "
,0.   %% no empty `.`
,00   %% first digit 0 can't have another number after it
```

We can just use `\d+\.?\d*` but I wouldn't recommend it.


As for the second terminal rule, I'm not if we could simplify the doable quote part.
I think `state` in `jison` are similar to `mode` in `chevrotain`, so when the use `"` it should enters a state where it should consume everything until it finds another `"`, but I don't know how to use the same rule to enter and exist mode.
But with modes the following code would be invalid (where is should be since `sankey` supposed to be `csv` like grammar):

```
sankey-beta
source",target,0
```

Reply:
I think we could change langium rules to this:

```langium
terminal SANKEY_LINK_VALUE returns number: /,"0|[1-9][0-9]*)(\.[0-9]+)?"/ | /,[\t ]*(0|[1-9][0-9]*)(\.[0-9]+)?)/;
terminal SANKEY_LINK_TARGET: /,"(""|[^"])+"/ | /,[^\n\r,]+/;
```

Reply:
It appears that it would throw a warning when using this approach:
````
src/language/sankey/sankey.langium:45:45 - Regular expression flags are only applied if the terminal is not a composition
src/language/sankey/sankey.langium:45:66 - Regular expression flags are only applied if the terminal is not a composition
```